src/handlers/assets.js

```diff
@@ -3,25 +3,69 @@ import {
   convertAmountToBalanceDisplay,
   convertRawAmountToDecimalFormat,
 } from '../helpers/utilities';
-import { erc20ABI } from '../references';
-import { web3Provider } from './web3';
+import {
+  ARBITRUM_ETH_ADDRESS,
+  erc20ABI,
+  ETH_ADDRESS,
+  MATIC_POLYGON_ADDRESS,
+  OPTIMISM_ETH_ADDRESS,
+} from '../references';
+import networkTypes from '@rainbow-me/helpers/networkTypes';
+
+export function isNativeAsset(address, network) {
+  /* 
+    TL;DR of this nasty logic
+    check if an asset is native depending on
+    the network & asset address:
+    - ETH address + L1
+    - MATIC address + Polygon
+    - ARBITRUM ETH address + Arbtrum One
+    - OPTIMISM ETH ADDRESS + Optimism Mainnet
+  */
+
+  if (
```

Comment:
note to self: can make a references map of network to native asset address and make it pretty